Compatibility update

jacobdavidson opened the pull request:
Package and associated syntax updates

jacobdavidson committed:
replaced outdated package hungarian

jacobdavidson committed:
xgboost to json file update

jacobdavidson committed:
xgboost update

jacobdavidson committed:
xgboost update

jacobdavidson committed:
revert

jacobdavidson committed:
Revert "revert"

This reverts commit 8b30b1af1f44e31ec962f06f7f596784a76e912f.

jacobdavidson committed:
load with XGBClassifier

jacobdavidson committed:
np int update

walachey commented on bb_tracking/models.py:
```diff
@@ -76,30 +76,56 @@ def to_dict(self):
-    
-    
+        # Check if the model is an instance of xgb.Booster or XGBClassifier/XGBRegressor
+        if isinstance(self, xgb.Booster):
```
This can never happen

walachey commented on bb_tracking/models.py:
```diff
@@ -76,30 +76,56 @@ def to_dict(self):
+            self.save_model(path)
+        else:
+            # Convert to Booster and save if it's an XGBClassifier/XGBRegressor
+            booster = self.get_booster()
```
This should probably be self.model.get_booster(), but it also disregards the axis_normalizers_min_ / max and diff_normalizer_min_ / max which are also parameters of the fit

walachey commented on bb_tracking/models.py:
```diff
@@ -76,30 +76,56 @@ def to_dict(self):
-    
+
     @classmethod
+    # def load(cls, path):
```
Commented out code should be removed (the history will always be available anyway)

walachey committed:
reverted np.int32 -> int (np.in32 is not deprecated and this code might use numba, so I'd rather be strict about the type); also use the full module path to refer to the linear_sum_assignment function, because it's only used once

walachey committed:
reverted np.int32 -> int (np.int32 is not deprecated and this code uses numba, so I'd rather be strict about the types); also use full path to refer to scipy.optimize.linear_sum_assignment, because it's only used once

jacobdavidson committed:
removed commented code

jacobdavidson committed:
updated models.py so only uses XGBClassifier

updated models.py so only uses XGBClassifier (built-in xgboost class) for loading and saving

walachey commented on bb_tracking/training/tracklet_data_generation.py:
```diff
@@ -16,10 +16,10 @@
-    with open(detection_model_path, "rb") as f:
-            detection_model = joblib.load(f)
+
+    # Load the detection model
```
Unhelpful comment :D (no blocker tho)
